### ISSUE TITLE
`import … hiding …` should be documented

### ISSUE THREAD

**Issue by mietek** [agda/agda#3927]
As seen in https://github.com/agda/agda/issues/3879#issuecomment-512353532.

Identifiers can be in three states:

- accessible unqualified;
- accessible fully-qualified;
- inaccessible.

Writing
```agda
import M hiding (foo)
```
makes `foo` inaccessible.

**Comment by Blaisorblade** [comment]
Indeed. To improve on that: IIUC `import M hiding (foo)` works for any `foo`, not just `pattern-synonym`. You just need it seldom — in cases like #3879 for pattern synonyms (and maybe constructors, if you want to resugaring to use only pattern synonyms instead of constructors?).

**Comment by mietek** [comment]
Thanks; updated my comment.

**Comment by UlfNorell** [comment]
Currently [the documentation says](https://agda.readthedocs.io/en/latest/language/module-system.html):

> It is also possible to attach modifiers to import statements, limiting or changing what names are visible from inside the module.

And modifiers are explain above. I think that's reasonably clear.

**Comment by mietek** [comment]
@UlfNorell: I don’t think it’s clear that `import M hiding (foo)` makes `foo` not accessible even as `M.foo`. Additionally, see https://github.com/agda/agda/issues/3933.

**Comment by andreasabel** [comment]
@UlfNorell wrote:
> Currently [the documentation says](https://agda.readthedocs.io/en/latest/language/module-system.html):
> 
> > It is also possible to attach modifiers to import statements, limiting or changing what names are visible from inside the module.
> 
> And modifiers are explain above. I think that's reasonably clear.

Unfortunately, the link given wasn't a permalink.

I think the issue is still open.  
Current docs: https://agda.readthedocs.io/en/v2.7.0.1/language/module-system.html#name-modifiers
We explain the directives `using/hiding/renaming` in connection with `open`, `open import`, and `module` aliases, but not with bare `import`.  E.g., when running into #7656, I couldn't find the specs for `import File using (foo)`.
- #7656

**Comment by andreasabel** [comment]
Agda dev 2025-04-30: 
One application of `import M hiding (foo)` is to hide the instance `foo` (because instances are considered even if in scope only qualified).
